src/taskpane/taskpane.ts

```diff
@@ -6,28 +6,32 @@
 /* global Office, OfficeExtension */
 
 import * as excel from "./excel";
+import * as outlook from "./outlook";
 import * as powerpoint from "./powerpoint";
 import * as word from "./word";
 
 export function writeDataToOfficeDocument(result: string[]): Promise<any> {
-  return new OfficeExtension.Promise(function(resolve, reject) {
+  return new Promise(function (resolve, reject) {
     try {
       switch (Office.context.host) {
         case Office.HostType.Excel:
           excel.writeDataToOfficeDocument(result);
           break;
+        case Office.HostType.Outlook:
+          outlook.writeDataToOfficeDocument(result);
+          break;
         case Office.HostType.PowerPoint:
           powerpoint.writeDataToOfficeDocument(result);
           break;
         case Office.HostType.Word:
           word.writeDataToOfficeDocument(result);
           break;
         default:
-          throw "Unsupported Office host application: This add-in only runs on Excel, PowerPoint, or Word.";
+          throw "Unsupported Office host application: This add-in only runs on Excel, Outlook, PowerPoint, or Word.";
       }
       resolve();
     } catch (error) {
       reject(Error("Unable to write data to document. " + error.toString()));
     }
   });
-}
+}
```

Comment:
It's good to have a newline here on the last line. Perhaps the editor you're using removed it. Not a big deal but you can see it in the diffs.

Reply:
added newline back it